File: LojaDeCarroWeb/src/app/features/veiculos/models/veiculo-edicao-response.ts
import { VeiculoImagemResponse } from
  './veiculo-detalhe-response';
import { StatusVeiculo } from
  './veiculo-response';

export interface VeiculoEdicaoResponse {
  id: number;
  placa: string;
  quilometragem: number;
  valor: number;
  motor: string;
  descricao: string | null;
  anoFabricacao: number;
  idCarroceria: number;
  idCor: number;
  idModelo: number;
  idCombustivel: number;
  idsOpcionais: number[];
  statusVeiculo: StatusVeiculo;
  imagens: VeiculoImagemResponse[];
}

File: LojadeCarro/src/main/java/com/javacar/lojadecarro/dto/response/VeiculoEdicaoResponse.java
package com.javacar.lojadecarro.dto.response;

import com.javacar.lojadecarro.enums.StatusVeiculo;
import io.swagger.v3.oas.annotations.media.Schema;

import java.math.BigDecimal;
import java.util.List;

@Schema(description = "Dados de um veículo para edição")
public record VeiculoEdicaoResponse(
        Long id,
        String placa,
        Integer quilometragem,
        BigDecimal valor,
        String motor,
        String descricao,
        Short anoFabricacao,
        Long idCarroceria,
        Long idCor,
        Long idModelo,
        Long idCombustivel,
        List<Long> idsOpcionais,
        StatusVeiculo statusVeiculo,
        List<VeiculoImagemResponse> imagens
) {
}
